File: Smart-Bus/Smart-Bus/Classes/Main/ZoneControl/Audio/View/AudioRecordControlButton/SHAudioRecordControlButton.m
//
//  SHAudioRecoredControlButton.m
//  Smart-Bus
//
//  Created by Mark Liu on 2017/9/16.
//  Copyright © 2017年 SmartHome. All rights reserved.
//

#import "SHAudioRecordControlButton.h"

@implementation SHAudioRecordControlButton

- (instancetype)initWithFrame:(CGRect)frame {
    
    if (self = [super initWithFrame:frame]) {
        
        [self setUi];
    }
    return self;
}

- (void)awakeFromNib {
    
    [super awakeFromNib];
    
    [self setUi];
}

- (void)setUi {

    self.contentHorizontalAlignment = UIControlContentHorizontalAlignmentCenter;
    self.titleLabel.textAlignment = NSTextAlignmentLeft;
    
    self.imageView.contentMode = UIViewContentModeScaleAspectFit;

    self.imageEdgeInsets = UIEdgeInsetsMake(0, -10, 0, 0);
}

 

@end

File: Smart-Bus/Smart-Bus/Classes/Main/ZoneControl/Audio/View/AudioRecordControlButton/SHAudioRecordControlButton.swift
//
//  SHAudioRecordControlButton.swift
//  Smart-Bus
//
//  Created by Apple on 2019/6/28.
//  Copyright © 2019 SmartHome. All rights reserved.
//

import UIKit

class SHAudioRecordControlButton: UIButton {

    override init(frame: CGRect) {
        super.init(frame: frame)
        
        setUi()
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        setUi()
    }

    private func setUi() {

        contentHorizontalAlignment = .center
        titleLabel?.textAlignment = .left

        imageView?.contentMode = .scaleAspectFit

        imageEdgeInsets = UIEdgeInsets(top: 0, left: -10, bottom: 0, right: 0)
    }

}
